src/easylink/utilities/validation_utils.py

```diff
@@ -14,45 +14,243 @@
 from pyarrow import parquet as pq
 
 
-def validate_input_file_dummy(filepath: str) -> None:
-    """Validates an input file to a dummy :class:`~easylink.step.Step`.
-
-    This function is intended to be used as the :attr:`~easylink.graph_components.InputSlot.validator`
-    for _all_ input data at every step in the dummy/:mod:`easylink.pipeline_schema_constants.development`
-    pipeline schema. It simply checks for supported file types as well as the presence
-    of required columns.
+def _read_file(filepath: str) -> pd.DataFrame:
+    """
+    Reads a file.
```

Comment:
nit: in all easylink docstrings, we put the one-line comment right next to the """, e.g. 
```
def _read_file():
    """Reads a file.

    Parameters...
    """